libraries/chain/combined_database.cpp

```diff
@@ -164,6 +164,10 @@ namespace eosio { namespace chain {
             options.create_if_missing = true; // Creates a database if it is missing
             options.level_compaction_dynamic_level_bytes = true;
             options.bytes_per_sync = cfg.persistent_storage_bytes_per_sync; // used to control the write rate of flushes and compactions.
+            options.allow_mmap_reads = true;
```

Comment:
I did not do microbenchmarks.  Considering contention should be low, it made sense to employ an adaptive spin lock instead of resorting to a full blown mutex right away every time.  As for the mmap read/writes, I will remove them.